src/IO/ReadHelpers.h

```diff
@@ -861,16 +863,26 @@ inline ReturnType readDateTimeTextImpl(time_t & datetime, ReadBuffer & buf, cons
             UInt8 month = (s[5] - '0') * 10 + (s[6] - '0');
             UInt8 day = (s[8] - '0') * 10 + (s[9] - '0');
 
-            UInt8 hour = (s[11] - '0') * 10 + (s[12] - '0');
-            UInt8 minute = (s[14] - '0') * 10 + (s[15] - '0');
-            UInt8 second = (s[17] - '0') * 10 + (s[18] - '0');
+            UInt8 hour = 0;
+            UInt8 minute = 0;
+            UInt8 second = 0;
+            ///simply determine whether it is YYYY-MM-DD hh:mm:ss or YYYY-MM-DD by the content of the tenth character in an optimistic scenario
+            if (s[10] == ' ')
```

Comment:
ISO-8601 prefers `T` as a separator.

Reply:
@yakov-olkhovskiy let's check if DateTime like 2022-04-03T01:02:03 is covered by tests.

Reply:
@alexey-milovidov I would say the whole parser kinda loose - we don't check for anything besides fifth symbol is not being a digit and now this delimiter...

Reply:
Yes, it is loose by intention, because the list of separators between date and time is open.

We should definitely support whitespace ` ` and `T`, but maybe also `.`, `:`, `;`, `+`, `-`...